Api/RoutedResource.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+/*
+ * This file is part of the DunglasApiBundle package.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Dunglas\ApiBundle\Api;
+
+/**
+ * {@inheritdoc}
+ *
+ * @author Jérémy Hubert <jeremy@tofusteak.fr>
+ */
+class RoutedResource extends Resource implements RoutedResourceInterface
```

Comment:
IMO it's not necessary to introduce a new class. The existing `Resource` class can implement the new  `RoutedResourceInterface`.